src/components/Table/GridTable.tsx

```diff
@@ -300,105 +303,77 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
     RowTuple<R>[],
     string[],
   ] = useMemo(() => {
-    function makeRowComponent(
+    const hasExpandableHeader = maybeSorted.some((row) => row.id === EXPANDABLE_HEADER);
+    const makeRowComponent = (
       row: GridDataRow<R>,
       level: number,
       isKeptSelectedRow: boolean = false,
       isLastKeptSelectionRow: boolean = false,
-    ): JSX.Element {
-      return (
-        <Row
-          key={`${row.kind}-${row.id}`}
-          {...{
-            as,
-            columns,
-            row,
-            style,
-            rowStyles,
-            columnSizes,
-            level,
-            getCount,
-            api,
-            cellHighlight: "cellHighlight" in maybeStyle && maybeStyle.cellHighlight === true,
-            omitRowHover: "rowHover" in maybeStyle && maybeStyle.rowHover === false,
-            sortOn,
-            hasExpandableHeader,
-            isKeptSelectedRow,
-            isLastKeptSelectionRow,
-          }}
-        />
-      );
-    }
+    ) => (
+      <Row
+        key={`${row.kind}-${row.id}`}
+        {...{
+          as,
+          columns,
+          row,
+          style,
+          rowStyles,
+          columnSizes,
+          level,
+          getCount,
+          api,
+          cellHighlight: "cellHighlight" in maybeStyle && maybeStyle.cellHighlight === true,
+          omitRowHover: "rowHover" in maybeStyle && maybeStyle.rowHover === false,
+          hasExpandableHeader,
+          isKeptSelectedRow,
+          isLastKeptSelectionRow,
+        }}
+      />
+    );
 
     // Split out the header rows from the data rows so that we can put an `infoMessage` in between them (if needed).
     const headerRows: RowTuple<R>[] = [];
     const expandableHeaderRows: RowTuple<R>[] = [];
     const totalsRows: RowTuple<R>[] = [];
     const visibleDataRows: RowTuple<R>[] = [];
     const keptSelectedRows: RowTuple<R>[] = [];
-    const filteredRowIds: string[] = [];
-
-    const hasExpandableHeader = rows.some((row) => row.id === EXPANDABLE_HEADER);
 
-    function visit([row, children]: ParentChildrenTuple<R>, level: number, visible: boolean): void {
-      visible && visibleDataRows.push([row, makeRowComponent(row, level)]);
-      // This row may be invisible (because it's parent is collapsed), but we still want
-      // to consider it matched if it or it's parent matched a filter.
-      filteredRowIds.push(row.id);
```

Comment:
This `visibleDataRows` calc was complicated by having to keep adding rows to `filteredRowIds`, even after we're in a collapsed/invisible section of the tree. So I moved the `filteredRowIds` calc elsewhere, and this got simpler.